### DISCUSSION TITLE
Support subject line inside of React Email components

### DISCUSSION THREAD

**Discussion by psirenny** [resend/react-email#1339] [Ideas]
### Goals

Be able to support the subject line inside of React Email components the same way that preview text is supported.

### Background

Preview text is already supported within React Email like this:

```tsx
<Html>
  <Head />
  <Preview>
    {`Welcome to Acme co ${firstName}. Please confirm your email address.`}
  </Preview>
  <Body>
    <Container>
      {/* … */ }
    </Container>
  </Body>
</Html>
```

It would be really neat if the subject line was also supported:

```diff
<Html>
  <Head />
+ <Subject>
+   {`Welcome to Acme co ${firstName}`}
+ </Subject>
  <Preview>
    {`Welcome to Acme co ${firstName}. Please confirm your email address.`}
  </Preview>
  <Body>
    <Container>
      {/* … */ }
    </Container>
  </Body>
</Html>
```

### Proposal

Like above ☝️. The render call can then be tweaked with:

```diff
import { MyTemplate } from './email';
import { render } from '@react-email/render';

const text = render(<MyTemplate />, {
  plainText: true,
+ subject: true,
});
```

or it could be altered to something like:

```tsx
import { MyTemplate } from './email';
import { render } from '@react-email/render';

const { bodyHtml, bodyText, subject } = render(<MyTemplate />);
```

Although, that's a breaking change and the variable names are subjective.

**Comment by gabrielmfern** [comment]
Wow, this is actually a really good idea for a new API actually. For some time we wanted to allow the `render` to generate assets along the HTML for the email and I think this can get us closer to a good way of doing this. Maybe something like:

```tsx
const { html, plainText, attachments, subject } = render(<MyTemplate />); 
```

**Reply by gabrielmfern** [reply]
The coolest thing would be doing any image generation or something similar, and then returning the attachment and referencing it in the HTML which would allow us to have something like a QR Code component or even a Chart component if we do it right.

**Reply by Pab450** [reply]
Any news on this system? We're really looking forward to this feature! Thanks for all your hard work!

**Comment by galkin** [comment]
@gabrielmfern, for migrate from SendGrid templates to react.email, i really need this feature.

**Comment by Syed-Umair** [comment]
I love this idea, I would like to contribute on this. If we haven't started already ❤️ .

**Comment by jorisvdz** [comment]
This would be a really helpful feature!
